treeherder/client/thclient/client.py

```diff
@@ -674,7 +678,10 @@ def __init__(
     def _get_uri(self, project, endpoint, data=None, oauth_key=None,
                  oauth_secret=None, method='GET'):
 
-        uri = '{0}://{1}/api/project/{2}/{3}/'.format(
+        if method == 'POST':
+            endpoint += '/'
```

Comment:
So I'm uncertain about this -- it's a valid assumption for what we're getting right now via the API, but may not be in the future. In my PR for perf data, I got around this by changing _get_uri not to add the trailing slash, but to add the slash to the various endpoints for posting:

https://github.com/mozilla/treeherder/pull/536/files#diff-8f20c700b32fc2a7be610d1089eb9eb4R566

I can't decide for certain which approach is better, though I think I learn towards using mine because it's a bit more flexible and feels less special-casey. Mauro, Vaibhav what do you think?